connectors/sources/google_drive.py

```diff
@@ -0,0 +1,638 @@
+            full_response (bool, optional): Specifies whether the response is paginated or not. Defaults to False.
+
+        Raises:
+            exception: A instance of an exception class.
+
+        Yields:
+            Dictionary: Response returned by the resource method.
+        """
+        retry_counter = 0
+        while True:
+            try:
+                async with Aiogoogle(
+                    service_account_creds=self.service_account_credentials
+                ) as google_client:
+                    drive_client = await google_client.discover(
+                        api_name=API_NAME, api_version=API_VERSION
+                    )
+                    if RUNNING_FTEST and not sub_method and GOOGLE_DRIVE_EMULATOR_HOST:
+                        logger.debug(
+                            f"Using the google drive emulator at {GOOGLE_DRIVE_EMULATOR_HOST}"
+                        )
+                        # Redirecting calls to fake Google Drive server for e2e test.
+                        drive_client.discovery_document["rootUrl"] = (
+                            GOOGLE_DRIVE_EMULATOR_HOST + "/"
+                        )
+                    resource_object = getattr(drive_client, resource)
+                    method_object = getattr(resource_object, method)
+                    if full_response:
+                        first_page_with_next_attached = (
+                            await google_client.as_service_account(
+                                method_object(**kwargs),
+                                full_res=True,
+                                timeout=DRIVE_API_TIMEOUT,
+                            )
+                        )
+
+                        async for page_items in first_page_with_next_attached:
+                            yield page_items
+                            retry_counter = 0
+                    else:
+                        if sub_method:
+                            method_object = getattr(method_object, sub_method)
+                        yield await google_client.as_service_account(
+                            method_object(**kwargs), timeout=DRIVE_API_TIMEOUT
+                        )
+                    break
+            except AttributeError as exception:
+                logger.error(
+                    f"Error occurred while generating the resource/method object for an API call. Error: {exception}"
+                )
+                raise exception
+            except HTTPError as exception:
+                log_message = f"Retry count: {retry_counter} out of {self.retry_count}. Response code: {exception.res.status_code} Exception: {exception}."
+                retry_counter = await self._retry(retry_counter, log_message)
+            except Exception as exception:
+                log_message = f"Retry count: {retry_counter} out of {self.retry_count}. Exception: {exception}."
+                retry_counter = await self._retry(retry_counter, log_message)
```

Comment:
The `def _retry(retry_counter, log_message)` performs a condition check if current `retry_counter` is bigger than max retries and if true it would re-raise the exception. 

That being said, the `@retryable` decorator seems to be cleaner approach and I will amend the code to use it.